lib/ES/Template.pm

```diff
@@ -73,6 +77,19 @@ my $Autosense_RE = qr{
         :(?:CONSOLE|AUTOSENSE|KIBANA):
     }xs;
 
+#===================================
+sub _build_js {
+#===================================
+    my ( $self ) = (@_);
+
+    if ($self->{compiled_js} == 0) {
```

Comment:
I think this doesn't work with `--all` because it [forks](https://github.com/elastic/docs/blob/master/lib/ES/Book.pm#L264). I ran this to build a preview:
```
./build_docs --all --keep_hash --target_repo git@github.com:elastic/built-docs.git --target_branch haywood_1001     --push --rebuild
```

And that emits `Building docs.js` super frequently. I think you should make `build_js` public and call it around [here](https://github.com/elastic/docs/blob/master/build_docs.pl#L105) and [here](https://github.com/elastic/docs/blob/master/build_docs.pl#L242). I hate leaking the abstraction when we don't strictly have to but I think that it is important not to build the file a bunch of times, especially in different processes in parallel. I think we're bound to half built files if we do that.